SharpMUSH.Implementation/Functions/MathFunctions.cs

```diff
@@ -102,17 +150,163 @@ public static ValueTask<CallState> Encrypt(IMUSHCodeParser parser, SharpFunction
 	[SharpFunction(Name = "fraction", MinArgs = 1, MaxArgs = 2, Flags = FunctionFlags.Regular | FunctionFlags.StripAnsi)]
 	public static ValueTask<CallState> Fraction(IMUSHCodeParser parser, SharpFunctionAttribute _2)
 	{
-		throw new NotImplementedException();
+		var args = parser.CurrentState.ArgumentsOrdered;
+		try
+		{
+			var value = decimal.Parse(ArgHelpers.EmptyStringToZero(MModule.plainText(args["0"].Message)));
+			var showWhole = args.Count == 2 && int.Parse(ArgHelpers.EmptyStringToZero(MModule.plainText(args["1"].Message))) != 0;
+			
+			var wholePart = Math.Truncate(value);
+			var fractionalPart = Math.Abs(value - wholePart);
+			
+			// If it's a whole number, just return it
+			if (fractionalPart < 0.000001m)
+			{
+				return ValueTask.FromResult<CallState>(((int)wholePart).ToString());
+			}
+			
+			// Convert fractional part to a fraction using Fractions library
+			var fraction = Fractions.Fraction.FromDecimal(fractionalPart);
+			
+			// Adjust for negative numbers
+			var numerator = fraction.Numerator;
+			var denominator = fraction.Denominator;
+			
+			if (value < 0 && wholePart == 0)
+			{
+				numerator = -numerator;
+			}
+			
+			// If we have a whole part and showWhole is true
+			if (Math.Abs(wholePart) >= 1 && showWhole)
+			{
+				return ValueTask.FromResult<CallState>($"{(int)wholePart} {numerator}/{denominator}");
+			}
+			// If we have a whole part but showWhole is false, add it to the fraction
+			else if (Math.Abs(wholePart) >= 1 && !showWhole)
+			{
+				numerator += (long)wholePart * denominator;
+				return ValueTask.FromResult<CallState>($"{numerator}/{denominator}");
+			}
+			// Just the fraction
+			else
+			{
+				return ValueTask.FromResult<CallState>($"{numerator}/{denominator}");
+			}
+		}
+		catch (Exception)
+		{
+			return ValueTask.FromResult<CallState>(Errors.ErrorNumbers);
+		}
 	}
 
 	[SharpFunction(Name = "inc", MinArgs = 1, MaxArgs = 1, Flags = FunctionFlags.Regular | FunctionFlags.StripAnsi)]
 	public static ValueTask<CallState> Inc(IMUSHCodeParser parser, SharpFunctionAttribute _2)
 		=> ArgHelpers.EvaluateDecimal(parser.CurrentState.ArgumentsOrdered, x => x + 1);
 
 	[SharpFunction(Name = "lmath", MinArgs = 2, MaxArgs = 3, Flags = FunctionFlags.Regular | FunctionFlags.StripAnsi)]
-	public static ValueTask<CallState> LMath(IMUSHCodeParser parser, SharpFunctionAttribute _2)
+	public static async ValueTask<CallState> LMath(IMUSHCodeParser parser, SharpFunctionAttribute _2)
 	{
-		throw new NotImplementedException();
+		await ValueTask.CompletedTask;
+		var args = parser.CurrentState.ArgumentsOrdered;
+		
+		var operation = MModule.plainText(args["0"].Message).ToLower();
+		var delimiter = args.Count == 3 ? args["2"].Message : MModule.single(" ");
+		var list = MModule.split2(delimiter, args["1"].Message).ToList();
+		
+		try
```

Comment:
What exception are you expecting here? Can this try-catch be removed in favor of safer calls?